test/api/support/ffactory.js

```diff
@@ -0,0 +1,98 @@
+const Promise = require("bluebird")
+const githubAPINocks = require("./githubAPINocks")
+
+const { Build, BuildLog, Site, User } = require("../../../api/modelss")
+
+const factory = (model, overrides) => {
+  const attributes = defaultAttributes[model.name].call(undefined, overrides)
+
+  return Promise.props(attributes).then(attributes => {
+    prepareForCreation(model, attributes)
+    return model.create(attributes)
+  })
+}
+
+const prepareForCreation = (model, attributes) => {
+  switch (model.name) {
+    case "Site":
+      githubAPINocks.webhook({
+        owner: attributes.owner,
+        repo: attributes.repository,
+      })
+      break
+  }
+}
+
+const buildAttributes = (overrides = {}) => {
+  let { user, site } = overrides
+
+  if (!user) {
+    user = factory(User)
+  }
+  if (!site) {
+    site = Promise.resolve(user).then(user => {
+      return factory(Site, { users: [user] })
+    })
+  }
+
+  return Object.assign({
+    site: site,
+    user: user
+  }, overrides)
+}
+
+const buildLogAttributes = ({ build, source, output } = {}) => ({
+  build: build || factory(Build),
+  source: source || "clone.sh",
+  output: output || "This is output from the build container",
+})
+
+const siteAttributes = (overrides = {}) => {
+  let { users } = overrides
+
+  if (users === undefined) {
+    users = Promise.all([factory(User)])
+  }
+
+  const repository = generateUniqueRepository()
+
+  return Object.assign({
+    owner: repository.owner,
+    repository: repository.name,
+    engine: "jekyll",
+    users: users
+  }, overrides)
+}
+
+let repositroyNameStep = 1
```

Comment:
I vote we fix this typo and approve this PR. There are more important things to do than removing the bluebird dependency; it's not related to the de-sails-i-fi-ca-tion.